UI enhancements for jetpack focus fullscreen overlay in landscape mode

AjeshRPai opened the pull request:
Part of #17333

This PR changes the following 
- Adds the landscape XML for Jetpack full-screen overlay, makes the content scrollable
- updates the Portrait XML to handle large fonts for accessibility, the content is made scrollable
- fixes Jetpack fullscreen overlay being shown on top of the already shown jetpack fullscreen overlay on rotation 

### To test:

Pre- Requisites 
Launch WP app -> Go to my site -> App settings -> Debug settings -> Enable jp_removal_one flag

###  Check if the fonts are displayed properly on Properly in Landscape and Portrait 
1. Launch the app -> Go to my site -> Access stats/reader/notification feature 
2. Verify that the UI is shown properly 

|Landscape|Portrait|
|---|---|
|![Screenshot_20221122_173149](https://user-images.githubusercontent.com/17463767/203309866-2c19bcd3-28aa-4b96-b750-3cc4b4539638.png)|![Screenshot_20221122_173214](https://user-images.githubusercontent.com/17463767/203309879-f30dd92b-f56e-40e1-abaf-55effba07213.png)|  

###  Check if the fonts are displayed properly on increasing the size of the fonts 
1. Go to system settings -> Display and font size -> Increase both display size and font size
2. Launch the app -> Go to my site -> Access stats/reader/notification feature 
3. Verify that the UI is shown properly


|Landscape(Scrollable)|Portrait|
|---|---|
|![Screenshot_20221122_173436](https://user-images.githubusercontent.com/17463767/203311563-80297aba-dd1c-4400-bcd3-8005f330e930.png)|![Screenshot_20221122_173422](https://user-images.githubusercontent.com/17463767/203311797-7b19d5ad-e78d-4c7b-9b0e-799cc935e512.png)|  


### Check if the multiple full screen overlay is displayed when rotated 
- Launch the app -> Go to my site -> Access stats/reader/notification feature 
- Rotate the device when overlay is shown 
- Verify that on dismissing the overlay, no other overlay is displayed below it.


## ⚠️  Merge Instructions 
- Revert the commit added for testing purposes https://github.com/wordpress-mobile/WordPress-Android/pull/17510/commits/6d092ae2fced9abf5e1cb6467aa3637d52d8f234
- Remove not read for merge label 
- Merge as normal 

## Regression Notes
1. Potential unintended areas of impact


6. What I did to test those areas of impact (or what existing automated tests I relied on)


7. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

AjeshRPai committed:
Updates: logic not to show the overlay on rotation

AjeshRPai committed:
Updates: portrait UI to handle big fonts

AjeshRPai committed:
Adds: landscape UI for full screen overlay

AjeshRPai committed:
Updates: the ui logic for showing bottom sheet properly

AjeshRPai committed:
Adds: logic to show the overlay everytime for testing purpose

AjeshRPai commented on WordPress/src/main/java/org/wordpress/android/ui/jetpackoverlay/JetpackFeatureRemovalOverlayUtil.kt:
```diff
@@ -27,12 +27,9 @@ class JetpackFeatureRemovalOverlayUtil @Inject constructor(
-                feature,
-                getCurrentPhasePreference()!!
-        )
+        return true
```
This change is for testing purposes only. This change will make sure that the overlay is shown everytime the features are accessed.

AjeshRPai committed:
Fixes: lint issues

osullivanchris commented:
Good work @AjeshRPai it looks pretty good considering the limitations. Here's some suggested tweaks/enhancements:

- Maybe the heading can have a max size. It already starts with quite large, legible type. So it might not need to scale equally with the subheading.
- On landscape mode could we remove the top padding above the logo/icons, so that we see at least some text before the fold? Here's how it looks to me. It would be better if the default scroll position could look like [your screenshot here](https://user-images.githubusercontent.com/17463767/203309866-2c19bcd3-28aa-4b96-b750-3cc4b4539638.png) where I assume you have scrolled down a bit.
![image](https://user-images.githubusercontent.com/50576199/203319607-e0f1731e-124d-45d7-b497-be10272edd2c.png)
- If I make the text and display large, and change the language to German, the button copy is getting cut off. Do we need a dynamic/set max button text size? Or could we change the copy in languages when it gets too long? 
![image](https://user-images.githubusercontent.com/50576199/203320230-edb324df-8cb6-492b-959c-2d97d40e4ae9.png)

AjeshRPai committed:
Updates: Ui with design review comments

AjeshRPai commented:
@osullivanchris Thanks chris for the review. 

I have made the following changes as per your review comments. 

>Maybe the heading can have a max size. It already starts with quite large, legible type. So it might not need to scale equally with the subheading.

1. I have changed the scaling of the heading to have a maximum size of 32 sp. I haven't changed the scaling of the caption text 


Screenshot of the device with max font size and display size 👇🏼 

![Screenshot_20221122_193453](https://user-images.githubusercontent.com/17463767/203335666-43045f93-ede6-443e-8eab-f481e6bd8ec5.png)

>On landscape mode could we remove the top padding above the logo/icons, so that we see at least some text before the fold? Here's how it looks to me. It would be better if the default scroll position could look like [your screenshot here](https://user-images.githubusercontent.com/17463767/203309866-2c19bcd3-28aa-4b96-b750-3cc4b4539638.png) where I assume you have scrolled down a bit.

Yep, I have decreased the top margin of the heading to 16dp from 24dp. Let me know if this looks better. Also I have adjusted the padding between the logo and the heading below it from 24 dp to 16dp. 

Screenshot of the device in landscape mode with padding adjusted 👇🏼 

![Screenshot_20221122_193842](https://user-images.githubusercontent.com/17463767/203336496-55cf999b-59a7-466d-bd53-279c01b1f831.png)


>If I make the text and display large, and change the language to German, the button copy is getting cut off. Do we need a dynamic/set max button text size? Or could we change the copy in languages when it gets too long?

I tested by setting the button size to a cap of 14 sp(default size is 12 sp I believe), and increasing the display size and font size and setting the language to german. The text still doesn't fit the button on Portrait mode. I don't think we can handle the transition + large font size on the button. I think the only solution is to change the copy in languages when it gets too long. Right now the max font size on the button is set to 14sp. 

Screenshot of the device in portrait mode with button text size set to 14 sp and display and font size set to maximum

![Screenshot_20221122_193453](https://user-images.githubusercontent.com/17463767/203337557-9e2557b4-d1a9-42b8-848f-68afcf4fb8dc.png)

I have pushed the above changes with the commit - [b0ff534](https://github.com/wordpress-mobile/WordPress-Android/pull/17510/commits/b0ff534e06e6f14a7e887ec42dc250a9b0a97e8b)

AjeshRPai commented:
hey @zwarm @JB184351 👋🏼 

Since @osullivanchris is doing the design review. Testing the following scenario by one of you should be enough to merge this PR. Please feel free to ignore this PR if there is 2 approvals. 

### Check if the multiple full screen overlay is displayed when rotated
- Launch the app -> Go to my site -> Access stats/reader/notification feature
- Rotate the device when overlay is shown
- Verify that on dismissing the overlay, no other overlay is displayed below it.

JB184351 commented:
> hey @zwarm @JB184351 👋🏼
> 
> Since @osullivanchris is doing the design review. Testing the following scenario by one of you should be enough to merge this PR. Please feel free to ignore this PR if there is 2 approvals.
> 
> ### Check if the multiple full screen overlay is displayed when rotated
> * Launch the app -> Go to my site -> Access stats/reader/notification feature
> * Rotate the device when overlay is shown
> * Verify that on dismissing the overlay, no other overlay is displayed below it.

Verified each of these with no issues, looks good to me!

osullivanchris commented:
@AjeshRPai Hey nice work, thanks for those improvements. Its looking better. 

On landscape on larger display sizes we're still not seeing much information. We could
- reduce the max font size even more specifically for landscape
- hide the logos when the display size is large, on landscape
Do you think it is worth further work?

Re; longer strings in other languages. It would be better in that case if we got a shorter string. I assume it would work if we said "Get Jetpack", "Get App". But given that is not our ideal copy, how would you propose we get that shorter title for e.g. German? Would we ask for a custom translation for German? Or could we have two strings, and get translations for both. The first string would be preferable (Get the Jetpack App), the second string would be a backup string (Get app) if the copy is too long in another language. Let me know if you have a process/method in mind here.

AjeshRPai committed:
Updates: ui based on review comments

AjeshRPai committed:
Merge branch 'trunk' into issue/17333-jetpack-focus-fullscreen-overlay-landscape-mode

# Conflicts:
#	WordPress/src/main/res/values/dimens.xml

AjeshRPai commented:
@osullivanchris 👋🏼 

I have
-  disabled the scaling of heading in landscape mode
- decreased the padding between heading and caption in landscape mode 

Commit - [bbb2d61](https://github.com/wordpress-mobile/WordPress-Android/pull/17510/commits/bbb2d61a7f79e0377af2cc6a4ec06c6c67193d3a)

I think hiding the logo is not worth it. Since the content is scrollable and most of the users won't be using max display size + font size in landscape mode, I think it's okay to ignore that corner case.

AjeshRPai commented:
Hey @osullivanchris 

I have created [this Issue](https://github.com/wordpress-mobile/WordPress-Android/issues/17521) for tracking the big fonts on the button. 

I will take it up later as a separate PR. I am merging this PR for now.
